### PR TITLE
fix: blockHeight, absoluteSlot and blockTime should actually be withi…

### PR DESCRIPTION
**Pull request by matiwinnetou**
…n type: long

### CODE REVIEW

File: backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java [hunk 1]
```diff
@@ -91,9 +94,15 @@ private TransactionContent convertToTransactionContent(TxInfo txInfo) {
         TransactionContent transactionContent = new TransactionContent();
         transactionContent.setHash(txInfo.getTxHash());
         transactionContent.setBlock(txInfo.getBlockHash());
-        transactionContent.setBlockHeight(txInfo.getBlockHeight());
-        transactionContent.setBlockTime(txInfo.getTxTimestamp());
-        transactionContent.setSlot(txInfo.getAbsoluteSlot());
+        if (txInfo.getBlockHeight() != null) {
```
Comment (matiwinnetou) [inline backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java:97]: Can we make assumption that any of these value will never be null?  I find my own defensive programming here a bit clunky.

Comment (satran004) [reply backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java:97]: @matiwinnetou Thanks for the PR.

These should not be null unless there is a defect on the provider's side. But I think we can keep this null check.

**Review by satran004 — APPROVED** [review]

**Review by edridudi — APPROVED** [review]
Will Change data types in Koios Java Client as well for the next version:
https://github.com/cardano-community/koios-java-client/issues/161

File: backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java (hunk 1 above)
Comment (matiwinnetou) [reply backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java:97]: ok, feel free to merge when you think this is fine and right time for it.

**Pull request merged**